NOTES.VALGRIND

```diff
@@ -62,9 +62,9 @@ INSTALL file in the root of the OpenSSL source tree.
 
 Example command line:
 
-   make test EXE_SHELL="`/bin/pwd`/util/shlib_wrap.sh valgrind --error-exitcode=1 --leak-check=full -q" OPENSSL_ia32cap=":0"
+   make test EXE_SHELL="`/bin/pwd`/util/wrap.pl valgrind --error-exitcode=1 --leak-check=full -q" OPENSSL_ia32cap=":0"
```

Comment:
I think this PR broke `EXE_SHELL` and that this documentation is not correct: if my perl understanding is not failing me, the first thing `wrap.pl` will do if `EXE_SHELL` is set is to prepend `EXE_SHELL` to `@cmd`. So if `EXE_SHELL` contains `wrap.pl` this will end in recursion: `wrap.pl wrap.pl wrap.pl wrap.pl ..... ${cmd}`.

Problem is that most sensible `EXE_SHELL` will need to start with `wrap.pl` otherwise `OPENSSL_ENGINES` and `OPENSSL_MODULES` won't be defined.

Also, if `EXE_SHELL` is defined, `shlib_wrap.sh` will be bypassed, so `LD_LIBRARY_PATH` will also be unset (but maybe this is by design.

Reply:
@levitte @mspncp can you look into this? I have a system where I might need to use EXE_SHELL heavily and I believe this change really broke things!

Reply:
Try this patch

``` diff
diff --git a/util/wrap.pl b/util/wrap.pl
index 1c3b4e7c29..4c3d4713f1 100755
--- a/util/wrap.pl
+++ b/util/wrap.pl
@@ -19,11 +19,7 @@ $ENV{OPENSSL_MODULES} = $std_providers
 my $use_system = 0;
 my @cmd;
 
-if (($ENV{EXE_SHELL} // '') ne '') {
-    # We don't know what $ENV{EXE_SHELL} contains, so we must use the one
-    # string form to ensure that exec invokes a shell as needed.
-    @cmd = ( join(' ', $ENV{EXE_SHELL}, @ARGV) );
-} elsif (-x $unix_shlib_wrap) {
+if (-x $unix_shlib_wrap) {
     @cmd = ( $unix_shlib_wrap, @ARGV );
 } else {
     # Hope for the best
```

See, looking back at shlib_wrap and opensslwrap, I cannot see why I added EXE_SHELL processing.  A lapse of judgment, perhaps?

Reply:
Testing, preliminary results seem promising!

Reply:
@levitte And also, mandatory huge thanks for looking quickly into it!

Reply:
See #11258